model_analyzer/record/types/perf_latency.py

```diff
@@ -92,4 +93,4 @@ def __sub__(self, other):
             of the inverted nature of latency (lower is better)
         """
 
-        return PerfLatency(value=(other.value() - self.value()))
+        return self.__class__(value=(other.value() - self.value()))
```

Comment:
If this works, perhaps we can provide a default implementation (the most common kind) in class `Record`?

Reply:
It wouldn't change anything for latency because it requires a specific implementation of a lot of these functions.

Reply:
Yes. I think `__eq__`, `__lt__`, `__add__` and `__sub__` functions can all be moved to `Record` class. Any implementation can override them if necessary.

Reply:
Exactly, maybe leave it as is for now then.